### PR TITLE
Add the changelog for the v1.1.0-rc.0 release

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

Add the changelog for the v1.1.0-rc.0 release. The release is tentatively planned for today, October 26th.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @kron4eg 
/hold
for discussion

### CODE REVIEW

**Review by kron4eg — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,50 @@
 # Changelog
 
+# [v1.1.0-rc.0](https://github.com/kubermatic/kubeone/releases/tag/v1.1.0-rc.0) - 2020-10-26
```
Comment (xmudrii) [inline CHANGELOG.md:3]: Please update the release date before merging the PR.

**Review by kron4eg — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (kron4eg) [reply CHANGELOG.md]: ✅

**Comment by kron4eg** [comment]
/hold cancel

addressed the review

**Pull request merged**